# Question 1

**Answer**: uv 0.9.5 (d5f39331a 2025-10-21)

# Question 2

**Answer**: sha256=NcvXVeYGFKansD_SnD6Vd57T6ZEzP36eWUd5eRbdLjc

# Question 3

In [15]:
import pickle

In [16]:
model_file = 'pipeline_v1.bin'

In [17]:
with open(model_file,'rb') as f_in:
    dv,model = pickle.load(f_in)

In [18]:
dv, model

(DictVectorizer(), LogisticRegression(solver='liblinear'))

In [19]:
record = {
    "lead_source": "paid_ads",
    "number_of_courses_viewed": 2,
    "annual_income": 79276.0
}

In [20]:
X = dv.transform([record])

In [21]:
model.predict_proba(X)[0,1]

np.float64(0.5336072702798061)

**Answer**: 0.5336072702798061

# Question 4

**Answer**: convertion_probability: 0.534

# Question 5

**Answer**: 181MB

# Question 6

**Answer**: couldn't do it =(